src/app/components/DynamicContentStyles.tsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import styled from 'styled-components';
+import { query } from '../navigation/selectors';
+
+// tslint:disable-next-line: variable-name
+export const WithStyles = styled.div`
+  ${(props: { styles: string }) => props.styles}
+`;
+
+interface DynamicContentStylesProps extends React.HTMLAttributes<HTMLDivElement> {}
+
+// tslint:disable-next-line: variable-name
+const DynamicContentStyles = ({ children, ...otherProps }: React.PropsWithChildren<DynamicContentStylesProps>) => {
+  const queryParams = useSelector(query);
+  const [styles, setStyles] = React.useState('');
+
+  React.useEffect(() => {
+    const cssfileUrl = queryParams['content-style'] as string | undefined;
```

Comment:
does this need the cast?

Reply:
you reacted +1 but the cast is still here, does that mean you do need it? i thought it was handled by the state types

Reply:
I probably didn't stage this file before committing, thanks